test/Microsoft.NET.TestFramework/ProjectConstruction/TestProject.cs

```diff
@@ -239,8 +239,15 @@ namespace {this.Name}
     public class {this.Name}Class
     {{
         public static string Name {{ get {{ return ""{this.Name}""; }} }}
-    }}
-}}";
+";
+                    foreach (var dependency in this.ReferencedProjects)
+                    {
+                        source += $"        public static string {dependency.Name} {{ get {{ return {dependency.Name}; }} }}" + Environment.NewLine;
```

Comment:
Or is this simply a stackoverflow waiting to happen?

---
In reply to: [107795515](https://github.com/dotnet/sdk/pull/1024#discussion_r107795515) [](ancestors = 107795515)